protocol-designer/src/pages/Designer/DeckSetup/DeckSetupTools.tsx

```diff
@@ -320,13 +326,20 @@ export function DeckSetupTools(props: DeckSetupToolsProps): JSX.Element | null {
     }
     if (selectedModuleModel != null) {
       //  create module
-      dispatch(
-        createModule({
-          slot,
-          type: getModuleType(selectedModuleModel),
-          model: selectedModuleModel,
-        })
-      )
+      const moduleType = getModuleType(selectedModuleModel)
+      // enforce gripper present in order to add plate reader
+      if (moduleType === ABSORBANCE_READER_TYPE && !isGripperAttached) {
+        makeSnackbar('Gripper required to add absorbance reader')
+        return
+      } else {
+        dispatch(
+          createModule({
+            slot,
+            type: moduleType,
+            model: selectedModuleModel,
+          })
+        )
+      }
```

Comment:
this is a bit more readable imo

```suggestion
    if (selectedModuleModel != null) {
      //  create module
      const moduleType = getModuleType(selectedModuleModel)
      // enforce gripper present in order to add plate reader
      if (moduleType === ABSORBANCE_READER_TYPE && !isGripperAttached) {
        makeSnackbar('Gripper required to add absorbance reader')
        return
      }
      
      
        dispatch(
          createModule({
            slot,
            type: moduleType,
            model: selectedModuleModel,
          })
        )
    }
```

Reply:
totally, don't need the else block if we're early returning